### PR TITLE
Profile issue : 93 show and update user profile

### PR DESCRIPTION
**Pull request by thanekarpooja27**
show and update user profile in go-backend

### CODE REVIEW

File: go-backend/db/mock.go [hunk 1]
```diff
@@ -39,3 +39,12 @@ func (m *DBMockStore) UpdateOrganization(ctx context.Context, org Organization,
 	args := m.Called(ctx, org, id)
 	return args.Get(0).(Organization), args.Error(1)
 }
+
+func (m *DBMockStore) GetUser(ctx context.Context, id int) (user User, err error) {
+	args := m.Called(ctx, id)
+	return args.Get(0).(User), args.Error(1)
+}
+func (m *DBMockStore) UpdateUser(ctx context.Context, usr User, id int) (updatedUser User, err error) {
```
Comment (iamanujvrma) [inline go-backend/db/mock.go:47]: for being consistent, always call it `user`, not `usr`

File: go-backend/db/user.go [hunk 2]
```diff
@@ -6,17 +6,127 @@ import (
 	logger "github.com/sirupsen/logrus"
 )
 
+const (
+	getUserQuery = `SELECT id,
```
Comment (iamanujvrma) [inline go-backend/db/user.go:10]: do we always need the `soft_delete` columns in get response ?

File: go-backend/db/user.go [hunk 3]
```diff
@@ -6,17 +6,127 @@ import (
...
+	updateUserQuery = `UPDATE users SET (
+		first_name,
+		last_name,
+		email,
+		display_name,
+		profile_image,
+		soft_delete,
+		role_id,
+		hi5_quota_balance,
+		soft_delete_by,
```
Comment (iamanujvrma) [inline go-backend/db/user.go:33]: I don't think we should allow soft delete columns to be set via update API, there will be a separate delete API later which will set these fields

File: go-backend/db/user.go [hunk 4]
```diff
@@ -6,17 +6,127 @@ import (
...
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error fetching user")
+		return
+	}
+	return
+}
+
+func (s *pgStore) UpdateUser(ctx context.Context, userProfile User, userID int) (updatedUser User, err error) {
+	var dbUser User
+	err = s.db.Get(&dbUser, getUserQuery, userID)
```
Comment (iamanujvrma) [inline go-backend/db/user.go:75]: why are we ignoring the error here?

File: go-backend/db/user.go [hunk 5]
```diff
@@ -6,17 +6,127 @@ import (
...
+		userProfile.SoftDeleteBy,
+		userProfile.SoftDeleteAt,
+		userID,
+	)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error updating user profile")
+		return
+	}
+
+	err = s.db.Get(&updatedUser, getUserQuery, userID)
```
Comment (iamanujvrma) [inline go-backend/db/user.go:96]: also handle error here too, what if someone in future adds code below this line?

File: go-backend/migrations/1587381324_create_users.up.sql [hunk 6]
```diff
@@ -1,4 +1,14 @@
 CREATE TABLE users (
-  name text,
-  age integer
```
Comment (iamanujvrma) [inline go-backend/migrations/1587381324_create_users.up.sql:3]: indentation issue

File: go-backend/db/mock.go (hunk 1 above)
Comment (thanekarpooja27) [reply go-backend/db/mock.go:47]: Okay I will do correct it.

File: go-backend/db/user.go (hunk 2 above)
Comment (thanekarpooja27) [reply go-backend/db/user.go:10]: not needed each time but to check the user is currently enabled or not is not necessary?

File: go-backend/db/user.go [hunk 7]
```diff
@@ -6,17 +6,123 @@ import (
 	logger "github.com/sirupsen/logrus"
 )
 
+const (
+	getUserQuery = `SELECT id,
```
Comment (iamanujvrma) [inline go-backend/db/user.go:10]: you should only allow to fetch user profile where soft_deleted is null. Means the user is not deleted. This goes for all queries which are made on user model

File: go-backend/db/user_test.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
+package db
```
Comment (iamanujvrma) [inline go-backend/db/user_test.go:1]: @thanekarpooja27 lets remove this integration test file, we don't need this anymore. Instead we will be writing test cases using https://github.com/DATA-DOG/go-sqlmock

But we can do that in a separate PR.

Comment (iamanujvrma) [reply go-backend/db/user_test.go:1]: only the db test case file
